tarantool/connection.py

```diff
@@ -897,10 +897,33 @@ def connect_basic(self):
         :meta private:
         """
 
-        if self.host is None:
-            self.connect_unix()
-        else:
+        if self.host is not None:
             self.connect_tcp()
+            return
+        if isinstance(self.port, int):
+            self.connect_socket_fd()
+            return
+        self.connect_unix()
+
+    def connect_socket_fd(self):
+        """
+        Establish a connection using an existing socket fd.
+
+        :raise: :exc:`~tarantool.error.NetworkError`
+
+        :meta private:
+        """
+
+        try:
+            # If old socket already exists - close it and re-create.
+            self.connected = True
+            if self._socket:
+                self._socket.close()
+            self._socket = socket.socket(fileno=self.port)
+            self._socket.settimeout(self.socket_timeout)
```

Comment:
Does the `fd` socket from tarantool is blocking?

Reply:
As I understood, there is no difference. If `self.socket_timeout` is None, the socket is put in blocking mode. If is not None, will be used non-blocking socket (similar to other connections).

Reply:
I suppose it is not expected that the fd socket may change the type from non-blocking to blocking.